elements_ephys/readers/openephys.py

```diff
@@ -0,0 +1,172 @@
+import pathlib
+import pyopenephys
+import numpy as np
+
+
+"""
+OpenEphys plug-in for Neuropixels recording saves data in binary format the following the directory structure:
+(https://open-ephys.github.io/gui-docs/User-Manual/Recording-data/Binary-format.html)
+
+Record Node 102
+-- experiment1 (equivalent to a Session)
+   -- recording1
+   -- recording2
+      -- continuous
+         -- Neuropix-PXI-100.0 (probe0 ap)
+         -- Neuropix-PXI-100.1 (probe0 lf)
+         -- Neuropix-PXI-101.0 (probe1 ap)
+         -- Neuropix-PXI-101.1 (probe1 lf)
+         ...
+      -- events
+      -- spikes
+      -- structure.oebin
+-- experiment 2
+   ...
+-- settings.xml
+-- settings2.xml
+...
+"""
+
+
+class OpenEphys:
+
+    def __init__(self, experiment_dir):
+        self.sess_dir = pathlib.Path(experiment_dir)
+
+        oe_file = pyopenephys.File(self.sess_dir.parent)  # this is on the Record Node level
+
+        # extract the "recordings" for this session
+        self.experiment = [experiment for experiment in oe_file.experiments
+                           if pathlib.Path(experiment.absolute_foldername) == self.sess_dir][0]
```

Comment:
thanks, updated to use `next()` now, no need for the try catch since the match is guaranteed to be found (otherwise this would have failed on the earlier steps)